src/components/dialogs/filter/expert/expert-filter-utils.ts

```diff
@@ -140,13 +165,19 @@ export function importExpertRules(
             // values is a Set on server side, so need to sort
             if (rule.dataType === DataType.NUMBER) {
                 return rule.values
-                    .map((value) => parseFloat(value as string))
+                    .map((value) => {
```

Comment:
You can keep the previous code and add a new map right after, then we avoid duplicate code.